ios/MaasNearbyTransit/Sources/MaasNearbyTransit/Components/NearbyTransitScheduleBadgesScrollView.swift

```diff
@@ -0,0 +1,214 @@
+import SwiftUI
+import Swappable
+
+private struct GradientScrollDataPreferenceKey: PreferenceKey {
+    typealias Value = ScrollViewData
+
+    static var defaultValue: ScrollViewData = .init(size: .zero, contentOffset: 0, contentSize: .zero)
+
+    static func reduce(value: inout ScrollViewData, nextValue: () -> ScrollViewData) {
+        value = nextValue()
+    }
+}
+
+private struct GradientBoundsScrollView<Content>: View where Content: View {
```

Comment:
This is could be pretty universal, probably deserves its own file and place :D

Reply:
Moving this in to separated pull request.